packages/core-data/src/entities.js

```diff
@@ -134,10 +134,10 @@ function* loadPostTypeEntities() {
 			},
 			mergedEdits: { meta: true },
 			getTitle( record ) {
-				if ( name === 'wp_template_part' || name === 'wp_template' ) {
+				if ( [ 'wp_template_part', 'wp_template' ].includes( name ) ) {
 					return startCase( record.slug );
 				}
-				return get( record, [ 'title', 'rendered' ], record.id );
+				return record?.title?.renderd || record?.title || record.id;
```

Comment:
This is the actual change that shows the title.

Reply:
I think that's a good fix. 

I think the main issue is that the shape of the entity record is different between fetching and edits. I believe that's because we can't push an update to `record.title.raw` in the Rest API, so this should do it IMO